OpenRA.Mods.Common/Effects/SpawnActorEffect.cs

```diff
@@ -0,0 +1,71 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Effects;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class SpawnActorEffect : IEffect
+	{
+		readonly Actor actor;
+		readonly CPos[] pathAfterSpawn;
+		readonly Activity activityAtDestination;
+		readonly IMove move;
+		int remainingDelay;
+
+		public SpawnActorEffect(Actor actor)
+			: this(actor, 0, new CPos[0], null) { }
+
+		public SpawnActorEffect(Actor actor, int delay)
+			: this(actor, delay, new CPos[0], null) { }
+
+		public SpawnActorEffect(Actor actor, int delay, CPos[] pathAfterSpawn, Activity activityAtDestination)
+		{
+			this.actor = actor;
+			remainingDelay = delay;
+			this.pathAfterSpawn = pathAfterSpawn.Where(p => p != actor.Location).ToArray();
```

Comment:
What does the `Where` here do? I can't find it in the old code.

Reply:
ReinforceGlobal uses the first CPos in the array as spawn position, so I'm removing it from the array to avoid an unnecessary move order to self.Location.